dozer-ingestion/src/connectors/object_store/table_reader.rs

```diff
@@ -77,38 +77,33 @@ impl<T: Clone + Send + Sync> TableReader<T> {
             .map_err(|e| TableReaderError(StreamExecutionError(e)))?;
 
         tokio::pin!(data);
-        loop {
-            let item = data.next().await;
-            if let Some(Ok(batch)) = item {
-                for row in 0..batch.num_rows() {
-                    let mut fields = vec![];
-                    for col in 0..batch.num_columns() {
-                        let column = batch.column(col);
-                        fields.push(map_value_to_dozer_field(
-                            column,
-                            &row,
-                            resolved_schema.field(col).name(),
-                        )?);
-                    }
-
-                    ingestor
-                        .write()
-                        .handle_message((
-                            (0_u64, idx),
-                            IngestionMessage::OperationEvent(Operation::Insert {
-                                new: Record {
-                                    schema_id: Some(SchemaIdentifier { id, version: 0 }),
-                                    values: fields,
-                                    version: None,
-                                },
-                            }),
-                        ))
-                        .unwrap();
-
-                    idx += 1;
-                }
-            } else {
-                break;
+
+        while let Some(Ok(batch)) = data.next().await {
+            for row in 0..batch.num_rows() {
+                let fields = batch
```

Comment:
Main impacting change being here, as this `iter() + collect()` approach will pre-allocate all the necessary memory for the resulting vector instead of starting with an empty vector that causes multiple re-allocations as elements are pushed to it.

Additionally, the `.columns()` method allows for direct access to the slice, so it's simpler code without extra overhead (maybe even slightly more performant)